Integrations/src/main/java/io/deephaven/integrations/python/PythonFunction.java

```diff
@@ -65,6 +65,17 @@ public PythonFunction(final PyObject pyObjectIn, final Class classOut){
         }
     }
 
+    /**
+     * Calls the function defined in pyCallable with the given args
+     *
+     * @param args arguments to pass to the Python function
+     * @return the result of the function call
+     */
+    public PyObject pyObjectApply(Object ... args) {
+        PyObject out = pyCallable.call("__call__", args);
+        return out;
+    }
```

Comment:
I'm looking at `Input` and `Future`, and they are convincing me that the way you have implemented this has problems.  Take this code to illustrate the problem:
```
PythonFunction<IndexSet> gatherCaller = new PythonFunction<>(inputs[i].getFunc(), PyObject.class);
```
Here, you are creating a `PythonFunction<IndexSet>` to make the code compile.  You are doing this because the one and only argument to `apply` is an `IndexSet`, but you aren't using the class in this way.  That results in bizarre code that is tough to reason about if you think carefully.  You  may need to revert `PythonFunction` and create a new class that does what you want... or consider other alternatives.